src/code/z_view.c

```diff
@@ -51,80 +51,88 @@ void View_Init(View* view, GraphicsContext* gfxCtx) {
     view->up.z = 0.0f;
     view->eye.z = -1.0f;
 
-    if (D_8012ABF0) {
-        if (D_8012ABF0 == 0) {}
+    if (sViewNotInitialized) {
+        if (sViewNotInitialized == 0) {}
         osSyncPrintf("\nview: initialize ---\n");
-        D_8012ABF0 = false;
+        sViewNotInitialized = false;
     }
 
     view->unk_124 = 0;
-    view->flags = 1 | 2 | 4;
+    view->dirtyFlags = VIEW_VIEWING | VIEW_VIEWPORT | VIEW_PROJECTION_PERSPECTIVE;
     View_InitDistortion(view);
 }
 
-void func_800AA358(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
+void View_LookAt(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
     if (eye->x == lookAt->x && eye->z == lookAt->z) {
         eye->x += 0.1f;
     }
 
     view->eye = *eye;
     view->lookAt = *lookAt;
     view->up = *up;
-    view->flags |= 1;
+    view->dirtyFlags |= VIEW_VIEWING;
 }
 
-void func_800AA3F0(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
+/*
+ * Unused. View_LookAt is always used instead. This version is similar but
+ * is missing the input sanitization and the update to the dirtyFlags.
+ */
+void View_LookAtUnsafe(View* view, Vec3f* eye, Vec3f* lookAt, Vec3f* up) {
     view->eye = *eye;
     view->lookAt = *lookAt;
     view->up = *up;
 }
 
 void View_SetScale(View* view, f32 scale) {
-    view->flags |= 4;
+    view->dirtyFlags |= VIEW_PROJECTION_PERSPECTIVE;
     view->scale = scale;
 }
 
 void View_GetScale(View* view, f32* scale) {
     *scale = view->scale;
 }
 
-void func_800AA460(View* view, f32 fovy, f32 near, f32 far) {
+void View_SetPerspective(View* view, f32 fovy, f32 near, f32 far) {
     view->fovy = fovy;
     view->zNear = near;
     view->zFar = far;
-    view->flags |= 4;
+    view->dirtyFlags |= VIEW_PROJECTION_PERSPECTIVE;
 }
 
-void func_800AA48C(View* view, f32* fovy, f32* near, f32* far) {
+void View_GetPerspective(View* view, f32* fovy, f32* near, f32* far) {
     *fovy = view->fovy;
     *near = view->zNear;
     *far = view->zFar;
 }
 
-void func_800AA4A8(View* view, f32 fovy, f32 near, f32 far) {
+void View_SetOrtho(View* view, f32 fovy, f32 near, f32 far) {
     view->fovy = fovy;
     view->zNear = near;
     view->zFar = far;
-    view->flags |= 8;
+    view->dirtyFlags |= VIEW_PROJECTION_ORTHO;
     view->scale = 1.0f;
 }
 
-void func_800AA4E0(View* view, f32* fovy, f32* near, f32* far) {
+/*
+ * Identical to View_GetPerspective, and never called.
+ * Named as it seems to fit the "set, get" pattern.
+ */
+void View_GetOrtho(View* view, f32* fovy, f32* near, f32* far) {
     *fovy = view->fovy;
     *near = view->zNear;
     *far = view->zFar;
 }
 
 void View_SetViewport(View* view, Viewport* viewport) {
     view->viewport = *viewport;
-    view->flags |= 2;
+    view->dirtyFlags |= VIEW_VIEWPORT;
 }
 
 void View_GetViewport(View* view, Viewport* viewport) {
     *viewport = view->viewport;
 }
 
-void func_800AA550(View* view) {
+void View_ApplyScissor(View* view) {
```

Comment:
```suggestion
void View_ApplyShrinkWindow(View* view) {
```
Scissor is the specific way a shrink-window is applied, but many sub-systems within the game can be scissored. There's an entire file called shrink_window.c that is used to determine how the window shrinks, and this is the function that applies it

Reply:
shrink_window comes from debug strings though so idk if we want to propagate it that much, there may be a better name.
But yeah scissoring is a low-level term so maybe there's a better higher level name to figure out for this (imo it can be left for later if we can't reach an agreement but I'm open to discussion)

What do you mean by "many sub-systems within the game can be scissored" though? which sub-systems?

Reply:
If it's just restricting drawing to an axis-aligned rectangle, then `clip` or `clipRect` is a common term.

Reply:
"clip" means something else at the low level too unfortunately, and it's even related to scissoring so would be a very confusing choice (unfortunately x2)
( http://n64devkit.square7.ch/pro-man/pro12/12-03.htm#01 )

Reply:
Summary of possible names with objections raised so far:

`Scissor`: is a low-level term
`shrink_window`: is from debug strings. Also I personally find it confusing; when I first saw "shrink window" I thought of the screen-shrinking effect when the bell rings in MM. Feels like a bit of an engrish-y name for scissor/clipRect.
`clip`: is a low-level term

I don't think `scissor` or `clip` are particularly confusing here. Scissor is indeed a low-level term but it means basically the same thing at the low level and here.  "Clip" is a low-level term but it's also very commonly used to refer to a rectangular region that drawing is restricted to by application code, to speed things up or to implement masking. It doesn't generally cause much confusion, I think. It's very commonly seen in high-level graphics APIs and is rarely called anything other than "clipping",

Reply:
to phrase this another way:
If you say `View_ApplyScissor` it implies just any old scissor value (would expect it to be a value given as an arg, for example)

where as `View_ApplyShrinkWindow` is saying that its applying the scissor with the value obtained from the shrink window system, which is what this does.

So i would probably lean toward naming it after the higher level system its interacting with

agreed though that shrink window is not the best name for that system, but it will change in the future when we actually start renaming og system names

Reply:
Ah, thanks for the explanation about the letterboxing. `shrink_window` is fine for me. I don't have a strong opinion on the name for this, anyway, so am happy to just change the PR when there's consensus.